modules/nf-core/bwa/mem/main.nf

```diff
@@ -32,7 +39,7 @@ process BWA_MEM {
         -t $task.cpus \\
         \$INDEX \\
         $reads \\
-        | samtools $samtools_command $args2 --threads $task.cpus -o ${prefix}.bam -
+        | samtools $samtools_command $args2 --reference ${fasta} --threads $task.cpus -o ${prefix}.${extension} -
```

Comment:
Can we condition the provision of `--reference` on a) the extension and b) the fasta actually being provided? I feel like a backwards-compatible usage without fastas is important to retain. Will need an error is someone requests a cram without providing fasta though.

Reply:
Done!